src/python/WMComponent/TaskArchiver/CleanCouchPoller.py

```diff
@@ -221,3 +227,31 @@ def cleanAllLocalCouchDB(self, workflowName):
         # other wise return True.
         return True
         
+    def cleanAlreadyArchivedWorkflows(self):
+        """
+        loop through the workflows in couchdb, if archived delete all the data in couchdb
+        """
+        
+        localWMStats = self.wmstatsCouchDB.getDBInstance()
+        options = {"group_level": 1, "reduce": True}
+        try:
+            results = localWMStats.loadView("WMStatsAgent", "allWorkflows", options = options)['rows']
+            requestNames = [x['key'] for x in results]
+            logging.info("There are %s workfows to check for archived status" % len(requestNames))
+            
+            workflowDict = self.centralRequestDBReader.getStatusAndTypeByRequest(requestNames)
+            numDeletedRequests = 0
```

Comment:
Just a minor suggestion, how about move this line to L236 and then move the return line (L249) after the try/except block?